doc/lua_api.txt

```diff
@@ -142,12 +141,25 @@ Mod directory structure
     |   `-- <custom data>
     `-- another
 
-
 ### modname
 The location of this directory can be fetched by using
 `minetest.get_modpath(modname)`.
 
+### mod.conf
+A key-value store of mod details.
+
+* `name` - the mod name. Allows Minetest to determine the mod name even if the folder is wrongly named.
+* `depends` - A comma separated list of dependencies. These are mods that must be loaded before this mod.
+* `optional_depends` - A comma separated list of optional dependencies. Like a dependency, but no error if the mod doesn't exist.
```

Comment:
When we already get the opportunity - `dependencies` and `optional_dependencies` as nouns would fit better into the other settings we have.